lib/state/ui/reducer.ts

```diff
@@ -1,12 +1,26 @@
 import { difference, union } from 'lodash';
-import { combineReducers } from 'redux';
+import { AnyAction, combineReducers } from 'redux';
 
 import * as A from '../action-types';
 import * as T from '../../types';
 
 const defaultVisiblePanes = ['editor', 'noteList'];
 const emptyList: unknown[] = [];
 
+const editorMode: A.Reducer<T.EditorMode> = (
+  state = 'edit',
+  action: AnyAction
```

Comment:
you shouldn't need to import `AnyAction` or provide the type to `action` as it's covered by `A.Reducer`

Reply:
The linter was yelling at me.

Reply:
<img width="912" alt="Screen Shot 2020-01-27 at 7 13 08 PM" src="https://user-images.githubusercontent.com/6817400/73224980-25b85480-4139-11ea-8f47-8dff3206f4db.png">

App.setEditorMode is not typed in action-types.ts.

Should we go ahead and type all the legacy "actions" out while we move everything over to redux?

Reply:
I don't know if it's worth typing all the existing actions. That would be a good PR to write and see if it's practical. I'm more comfortable leaving in the type errors though than doing something we wouldn't want to do in order to satisfy the linter.

Reply:
I removed the type

Reply:
I removed the type, will explore creating a PR, but I think that I'll be through this sooner than later making that work obsolete.